pkg/templates/templates.go

```diff
@@ -481,9 +491,11 @@ func (t *TemplateResolver) processForDataTypes(str string) string {
 
 	d1 := regexp.QuoteMeta(t.config.StartDelim)
 	d2 := regexp.QuoteMeta(t.config.StopDelim)
-	re := regexp.MustCompile(
-		`:\s+(?:[\|>]-?\s+)?(?:'?\s*)(` + d1 + `.*\|\s*(?:toInt|toBool|toLiteral).*` + d2 + `)(?:\s*'?)`,
-	)
+	expression := `:\s+(?:[\|>]-?\s+)?(?:'?\s*)(` + d1 + `.*\|\s*(?:toInt|toBool|toLiteral).*` + d2 + `)(?:\s*'?)`
+	if strings.Contains(str, "copyConfigMapData") || strings.Contains(str, "copySecretData") {
+		expression = `:\s+(?:[\|>]-?\s+)?(?:'?\s*)(` + d1 + `.*(?:copyConfigMapData|copySecretData).*` + d2 + `)(?:\s*'?)`
+	}
+	re := regexp.MustCompile(expression)
```

Comment:
Does this work?
```suggestion
	expression := `:\s+(?:[\|>]-?\s+)?(?:'?\s*)(` + d1 + `.*\|\s*(?:toInt|toBool|toLiteral|copyConfigMapData|copySecretData).*` + d2 + `)(?:\s*'?)`
	re := regexp.MustCompile(expression)
```

Reply:
The regular expressions are a little different, so I don't have one that works for both scenarios at the moment.

Reply:
Ah, I see. How about this?
```suggestion
	expression := `:\s+(?:[\|>]-?\s+)?(?:'?\s*)(` + d1 + `(?:.*\|\s*(?:toInt|toBool|toLiteral)|(?:.*(?:copyConfigMapData|copySecretData))).*` + d2 + `)(?:\s*'?)`
	re := regexp.MustCompile(expression)
```